src/lib/process.ts

```diff
@@ -0,0 +1,118 @@
+import { EventListener } from "./events";
+
+const RunService = game.GetService("RunService");
+
+export default function processHandler() {
+	const TPS = 20; // TPS to be replaced with value from tina.yaml
+	const processes = new Map<string, Process>();
+	const scheduler = new ProcessScheduler(TPS);
+
+	return (name: string) => {
+		if (processes.has(name)) {
+			return processes.get(name)!;
+		}
+		return new Process(name, scheduler);
+	};
+}
+
+/**
+ *
+ */
+export class Process extends EventListener<[]> {
+	public isSuspended = false;
+	public suspensionTime = -1;
+	public name: string;
+
+	constructor(name: string, private ticker: ProcessScheduler) {
+		super();
+		this.name = name;
+	}
+
+	public resume() {
+		this.isSuspended = false;
+		this.ticker.addProcess(this);
+	}
+
+	public suspend(seconds?: number) {
+		if (seconds === undefined) {
+			this.ticker.removeProcess(this);
+			return;
+		}
+
+		this.suspensionTime = math.floor(seconds * this.ticker.TPS);
+		this.isSuspended = true;
+	}
+}
+
+/**
+ *
+ */
+export class ProcessScheduler {
```

Comment:
The process scheduler should be made natively here, so let's not export the ProcessScheduler, let's instead export an instance of it.